### PR TITLE
fix the active learning modules in context to refinery workflow

### PR DESCRIPTION
**Pull request by divyanshukatiyar**
Changes:
- call abstract method `predict_prob` -> `predict_proba`
- `self.model.predict` -> `self.model.predict_proba`
- added the number of iterations as a user defined variable `YOUR_ITERATIONS`

### CODE REVIEW

File: classifiers/active_learners/random_search/code_snippet.md [hunk 1]
```diff
@@ -33,5 +34,5 @@ class MyRandom(LearningClassifier):
     )
 
     def predict_proba(self, embeddings):
-        return self.model.predict(embeddings)
+        return self.model.predict_proba(embeddings)
```
Comment (LeonardPuettmann) [inline classifiers/active_learners/random_search/code_snippet.md:37]: Why is .predict replaced with .predict_proba?

Comment (divyanshukatiyar) [reply classifiers/active_learners/random_search/code_snippet.md:37]: They have different dimensionality and are quite different in the context of refinery. `predict_proba` will return a 2D array with each row representing the samople in the input data, and each column represents the predicted probability. 
`predict` returns a 1-D array, where each element represents the predicted label for a set of input data.

**Comment by JWittmeyer** [comment]
Tested a few and the ones that I tested seem to run good.
Some of them (e.g. grid search) take quite the time (> 10 min for 3k records) -> don't know if that's a problem. Atm I don't think it is.


Two things I noticed:
1. I would expect all active learners to have this section in the variables `YOUR_LABELS: List[str] = None # optional, you can specify a list to filter the predictions` and the variable used later in the code.
2. The Train test split as variable is irritating for me since I can't do anything with it (was also a comment in the notion list but I don't know what the final result there is)

- [ ] resolved

**Comment by divyanshukatiyar** [comment]
I have made the changes addressed, apologies for the second point, somehow I skipped it. 
Also, the grid search is the most time taking algorithm by logic, so I don't know if the AL is performing at its supposed speed or something else.

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**